File: examples/integrity/wait_anchor/wait_anchor.py
import os
import bloock
from bloock.client.integrity import IntegrityClient
from bloock.client.record import RecordClient

if __name__ == "__main__":
    # we set the API key and create a client
    bloock.api_key = os.environ["API_KEY"]
    record_client = RecordClient()
    integrity_client = IntegrityClient()

    record = record_client.from_string("Hello world").build()
    records = [record]

    send_receipts = integrity_client.send_records(records)

    # Once we sent a record, we can wait for it's anchor
    print("Waiting for anchor...")

    # we can optionally specify a timeout (if not set, default is 120000)
    anchor = integrity_client.wait_anchor(send_receipts[0].anchor, timeout=120000)
    print("Done!")

File: examples/integrity/wait_anchor/wait_anchor.ts
import { Bloock, IntegrityClient, RecordClient } from '@bloock/sdk';

try {
  // we set the API key and create a client
  Bloock.setApiKey(process.env['API_KEY']);
  const recordClient = new RecordClient();
  const integrityClient = new IntegrityClient();

  const record = await recordClient.fromString('Hello world').build();
  const records = [record];

  const sendReceipts = await integrityClient.sendRecords(records);
  // Once we sent a record, we can wait for it's anochor
  console.log('Waiting for anchor...');
  // we can optionally specify a timeout (if not set, default is 120000)
  const _anchor = await integrityClient.waitAnchor(
    sendReceipts[0].anchor,
    120000
  );
  console.log('Done!');
} catch (e) {
  console.log(e);
}
